### PR TITLE
Bot improvements - CLA checks, killing old jobs, nagging about form

### PR DESCRIPTION
**Pull request by felixmulder**
Bot improvements include:

- Deleting old jobs from PR, only keep the latest drone build
- Check CLA and only approve last commit
- Add some greetings to opened PRs helping people on where to sign CLA etc

### CODE REVIEW

**Comment by felixmulder** [comment]
@dotty-bot: could you recheck this please?

**Comment by DarkDimius** [comment]
> Add some greetings to opened PRs helping people on where to sign CLA etc

would it be only for new commiters?

**Comment by felixmulder** [comment]
@dotty-bot: hey, need you to recheck this again please.

**Comment by felixmulder** [comment]
@dotty-bot: do you always say the same thing? Please recheck.

**Comment by felixmulder** [comment]
@dotty-bot: I don't like you.

File: bot/src/dotty/tools/bot/PullRequestService.scala [hunk 1]
```diff
@@ -170,20 +182,258 @@ trait PullRequestService {
...
+
+  def cancelBuilds(commits: List[Commit])(implicit client: Client): Task[Boolean] =
+    Task.gatherUnordered {
+      commits.map { commit =>
+        for {
+          statuses    <- getStatus(commit, client)
+          cancellable =  statuses.filter(status => status.state == "pending" && status.context == droneContext)
+          runningJobs =  cancellable.map(_.target_url.split('/').last.toInt)
+          cancelled   <- Task.gatherUnordered(runningJobs.map(Drone.stopBuild(_, droneToken)))
+        } yield cancelled.foldLeft(true)(_ == _)
```
Comment (OlivierBlanvillain) [inline bot/src/dotty/tools/bot/PullRequestService.scala]: `.forall(identity)` :)

File: bot/src/dotty/tools/bot/PullRequestService.scala [hunk 2]
```diff
@@ -170,20 +182,258 @@ trait PullRequestService {
...
+      req <- post(issueCommentsUrl(issueComment.issue.number)).withAuth(githubUser, githubToken)
+      _   <- client.fetch(req.withBody(comment.asJson))(Task.now)
+      res <- Ok("Delivered could not understand comment")
+    } yield res
+  }
+
+  def extractMention(body: String): Option[String] =
+    body.lines.find(_.startsWith("@dotty-bot:"))
+
+  /** TODO: The implementation here could be quite elegant if we used a trie instead */
```
Comment (OlivierBlanvillain) [inline bot/src/dotty/tools/bot/PullRequestService.scala]: I think you can turn this into a `Note:`...

**Comment by felixmulder** [comment]
Made the suggested changes (mostly). As this doesn't do anything inside the compiler, I cancelled the CI jobs.

**Comment by felixmulder** [comment]
thanks @OlivierBlanvillain !

File: bot/src/dotty/tools/bot/PullRequestService.scala [hunk 3]
```diff
@@ -207,7 +204,7 @@ trait PullRequestService {
         => false
       }
 
-      wrongTense || firstLine.last == '.' || firstLine.length > 100
+      wrongTense || firstLine.last == '.' || firstLine.length > 80
```
Comment (OlivierBlanvillain) [inline bot/src/dotty/tools/bot/PullRequestService.scala]: 72

**Comment by felixmulder** [comment]
I'm telling people that it's 72 (see bot message), but come on - let's
allow a bit of slack ;)

**Comment by smarter** [comment]
It lives!

**Pull request merged**